### DISCUSSION TITLE
Looking for help for building bar chart

### DISCUSSION THREAD

**Discussion by wadouk** [gampleman/elm-visualization#131] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/787448/139089937-67f44256-342a-45d4-a6b8-9c99e4ace3de.png)

already succeed with the [sample](https://code.gampleman.eu/elm-visualization/BarChart/) to change the axis orientation
https://ellie-app.com/fGPhMNMHGTma1

but try to change the width or height of the graph and stuff goes wrong

the height of the graph should be variable to the number of series to keep the width of one value

Thanks for help

**Comment by gampleman** [comment, marked as answer]
You're using a BandScale which is designed to do the opposite: to keep the overall space constant and divide the space equally between bands (with optional padding and spacing).

If you don't want that behavior, than the alternative is to not use a scale at all, just pick a width and use multiplication, i.e. something like this:

```elm

barHeight = 40

view bars = 
    svg [ viewBox 0 0 w (barHeight * List.length bars + somePadding) ] 
        [ ...
        , List.indexedMap (\index bar -> rect [ y (index * barHeight), ...] ) bars
        ]
```

**Comment by wadouk** [comment]
@gampleman thanks. It's so obvious. 

Any advice for the scale with ticks all under the chart?

**Reply by gampleman** [reply]
Not sure what you mean? Can you explain what you want to achieve and where are you stuck?

**Reply by wadouk** [reply]
On the mokup, there is no horizontal line to show the scale, and the ticks that show the numbers are drawn all the height of the chart. 
Not already dig so much to see how to make it. 
Seen they are advanced way to draw scales but don't understand how to handle the tick size (the vertical light grey lines). Will try tomorrow (CEST).

**Reply by wadouk** [reply]
Found what I want in the box [plot sample](https://elm-visualization.netlify.app/boxplot/) [line 225](https://ellie-app.com/fH7L8Lj2x7Ca1)
